linode/nb/framework_resource_schema.go

```diff
@@ -48,6 +48,10 @@ var frameworkResourceSchema = schema.Schema{
 			Computed: true,
 			Default:  int64default.StaticInt64(0),
 		},
+		"firewall_id": schema.Int64Attribute{
+			Description: "ID for the firewall you'd like to use with this NodeBalancer.",
+			Optional:    true,
+		},
```

Comment:
In order to prevent user confusion (expecting a change, no change made), we might want to make this field either `ForceNew` or implement read/update logic.